---
title: "Gene Set Analysis"
author: "Harlan Gillespie"
date: "21/10/2021"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Dependencies

```{r Dependencies, message=FALSE}
library(AnnotationHub)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(ChIPpeakAnno)
library(org.Hs.eg.db)
```


## Load the data

The 'genes_logFC.tsv' file created in Statistical Analysis previously is read and stored as a data.frame.

```{r}
df = read.table(file = 'genes_logFC.tsv', sep = '\t', header = TRUE)
```

Next, the fetal and adult upregulated genes are separated. Adult sample was used as a a control so upregulated fetal genes are labeled "up" and upregulated adult genes are labeled "down".

```{r}
up = df[df$logFC > 0,]
down = df[df$logFC < 0,]
```

## Get Annotations

Using AnnotationHub, the data from the roadmap epigenomics project containing locations of promoter associated histone modification H3K4me3 narrow peaks in human brain samples for adult and fetal samples.
```{r}
ah = AnnotationHub()
ah = subset(ah, species == "Homo sapiens")
qH3K4 = query(ah, "H3K4me3")
qbrain = query(qH3K4, c("brain", "narrowPeak"))
qfetal = query(qbrain, "fetal")

```

We are looking for a consolidated dataset. As 4 out of 6 of our samples are male and sex was not determined in PCA to be a significant covariate, we will use only the male data, E081.

```{r}
qfetal$tags[1]
fetal.brain.gr = qfetal[[1]]
```

E073 sample was chosen for the adult dataset as it represents H3K4me3 narrow peaks for the dorsolateral prefrontal cortex in ~80 year old males. Our samples are from people aged 36-42 years however this should suffice for comparison to fetal samples.

```{r}
qbrain$tags[7]
adult.brain.gr = qbrain[[7]]
```

Now to find adult and fetal peaks in liver tissue which will be used as a control. E066 was chosen as it is consolidated, narrowPeak data from liver samples.

```{r}
qliver = query(qH3K4, c("liver", "GRanges", "narrowPeak"))
qliver$tags[1]
liver.gr = qliver[[1]]
```



## Finding hg19 promoters that overlap with H3K4me3 peaks and annotation

```{r}
refseq <- ah[ah$genome == "hg19" & ah$title == "RefSeq Genes"][[1]]
prom = promoters(refseq)
prom = reduce(prom, ignore.strand = TRUE)
```

Find overlaps

```{r}
adult.brain.prom = subsetByOverlaps(adult.brain.gr, prom)
fetal.brain.prom = subsetByOverlaps(fetal.brain.gr, prom)
liver.prom = subsetByOverlaps(liver.gr, prom)
```

Preparation of annotation data.

```{r}
txdb = TxDb.Hsapiens.UCSC.hg19.knownGene
annoData = toGRanges(txdb, feature = "gene", single.strand.genes.only = FALSE)
annoData = keepStandardChromosomes(annoData, pruning.mode = "coarse")
annoData = dropSeqlevels(annoData, "chrM")
```

Annotating promoter GRanges.

```{r}
adult.prom.anno <- annotatePeakInBatch(adult.brain.prom, 
                                     AnnotationData=annoData, 
                                     output="nearestBiDirectionalPromoters",
                                     bindingRegion=c(-2000, 500))
fetal.prom.anno <- annotatePeakInBatch(fetal.brain.prom, 
                                     AnnotationData=annoData, 
                                     output="nearestBiDirectionalPromoters",
                                     bindingRegion=c(-2000, 500))
liver.prom.anno <- annotatePeakInBatch(liver.prom, 
                                     AnnotationData=annoData, 
                                     output="nearestBiDirectionalPromoters",
                                     bindingRegion=c(-2000, 500))
```


```{r}
adult.prom.genes = as.list(mcols(adult.brain.anno)[7])$feature
fetal.prom.genes = as.list(mcols(fetal.brain.anno)[7])$feature
liver.prom.genes = as.list(mcols(liver.anno)[7])$feature
```


```{r}
fetal.exp.genes = list(up[,1])[[1]]
adult.exp.genes = list(down[,1])[[1]]
```


## Calculating odds ratios

Creating function to calculate odds-ratios

```{r}
odds_ratio = function(prom, exp, entrez.total) {
  both = length(intersect(prom, exp))
  prom.only = length(setdiff(prom, exp))
  exp.only = length(setdiff(exp, prom))
  neither = entrez.total - sum(both, prom.only, exp.only)
  OR = (both/neither)/(exp.only/prom.only)
  return(OR)
}
```

Now, we can use the odds_ratio function to find our six odds ratios to fill a 2x3 matrix representing comparisons between the two expressed gene sets and the three H3K4me3 promoter genesets.

```{r}
ORMat <- matrix(0, ncol = 2, nrow = 3)
colnames(ORMat) <- c("fetal.exp", "adult.exp")
rownames(ORMat) <- c("fetal.prom", "adult.prom", "liver.prom")
```



```{r}
et = 87000
ORMat[1,1] = odds_ratio(fetal.prom.genes, fetal.exp.genes, et)
ORMat[1,2] = odds_ratio(fetal.prom.genes, adult.exp.genes, et)
ORMat[2,1] = odds_ratio(adult.prom.genes, fetal.exp.genes, et)
ORMat[2,2] = odds_ratio(adult.prom.genes, adult.exp.genes, et)
ORMat[3,1] = odds_ratio(liver.prom.genes, fetal.exp.genes, et)
ORMat[3,2] = odds_ratio(liver.prom.genes, adult.exp.genes, et)
ORMat
```



