---
title: "Gene Set Analysis"
author: "Harlan Gillespie"
date: "21/10/2021"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Dependencies

```{r Dependencies, message=FALSE}
library(AnnotationHub)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(ChIPpeakAnno)
library(org.Hs.eg.db)
```


## Load the data

The 'genes_logFC.tsv' file created in Statistical Analysis previously is read and stored as a data.frame.

```{r}
df = read.table(file = 'genes_logFC.tsv', sep = '\t', header = TRUE)
```

Next, the fetal and adult upregulated genes are separated. Adult sample was used as a a control so upregulated fetal genes are labeled "up" and upregulated adult genes are labeled "down".

```{r}
up = df[df$log2FoldChange > 2 & df$padj < 0.05,]
down = df[df$log2FoldChange < -2 & df$padj < 0.05,]
fetal.exp.genes = as.character(up[,1])
fetal.exp.genes = fetal.exp.genes[!is.na(fetal.exp.genes)]
adult.exp.genes = na.omit(as.character(down[,1]))
adult.exp.genes = adult.exp.genes[!is.na(adult.exp.genes)]
```

## Get Annotations

Using AnnotationHub, the data from the roadmap epigenomics project containing locations of promoter associated histone modification H3K4me3 narrow peaks in human brain samples for adult and fetal samples.
```{r}
ah = AnnotationHub()
ah = subset(ah, species == "Homo sapiens")
qH3K4 = query(ah, "H3K4me3")
qbrain = query(qH3K4, c("brain", "narrowPeak"))
qfetal = query(qbrain, "fetal")

```

We are looking for a consolidated dataset. As 4 out of 6 of our samples are male and sex was not determined in PCA to be a significant covariate, we will use only the male data, E081.

```{r}
qfetal$tags[1]
fetal.H3K4.gr = qfetal[[1]]
```

E073 sample was chosen for the adult dataset as it represents H3K4me3 narrow peaks for the dorsolateral prefrontal cortex in ~80 year old males. Our samples are from people aged 36-42 years however this should suffice for comparison to fetal samples.

```{r}
qbrain$tags[7]
adult.H3K4.gr = qbrain[[7]]
```

Now to find adult and fetal peaks in liver tissue which will be used as a control. E066 was chosen as it is consolidated, narrowPeak data from liver samples.

```{r}
qliver = query(qH3K4, c("liver", "GRanges", "narrowPeak"))
qliver$tags[1]
liver.H3K4.gr = qliver[[1]]
```
```{r}
adult.H3K4.gr = keepStandardChromosomes(adult.H3K4.gr, pruning.mode = "coarse")
adult.H3K4.gr = dropSeqlevels(adult.H3K4.gr, c("chrM", "chrX", "chrY"), pruning.mode = "coarse")

fetal.H3K4.gr = keepStandardChromosomes(fetal.H3K4.gr, pruning.mode = "coarse")
fetal.H3K4.gr = dropSeqlevels(fetal.H3K4.gr, c("chrM", "chrX", "chrY"), pruning.mode = "coarse")

liver.H3K4.gr = keepStandardChromosomes(liver.H3K4.gr, pruning.mode = "coarse")
liver.H3K4.gr = dropSeqlevels(liver.H3K4.gr, c("chrM", "chrX", "chrY"), pruning.mode = "coarse")
```



## Finding hg19 promoters of fetal and adult DEGs


```{r}
txdb = TxDb.Hsapiens.UCSC.hg19.knownGene
all.prom = promoters(genes(txdb))

fetal.exp.genes = fetal.exp.genes[!fetal.exp.genes %in% setdiff(fetal.exp.genes, all.prom$gene_id)]
adult.exp.genes = adult.exp.genes[!adult.exp.genes %in% setdiff(adult.exp.genes, all.prom$gene_id)]

fetal.prom = all.prom[fetal.exp.genes]
adult.prom = all.prom[adult.exp.genes]

fetal.prom = keepStandardChromosomes(fetal.prom, pruning.mode = "coarse")
fetal.prom = dropSeqlevels(fetal.prom, c("chrM", "chrX", "chrY"), pruning.mode = "coarse")

adult.prom = keepStandardChromosomes(adult.prom, pruning.mode = "coarse")
adult.prom = dropSeqlevels(adult.prom, c("chrM", "chrX", "chrY"), pruning.mode = "coarse")
```


## Calculating odds ratios

Creating function to calculate odds-ratios

```{r}
odds_ratio = function(H3K4, prom, all.prom) {
  prom = reduce(prom, ignore.strand = TRUE)
  H3K4 = reduce(H3K4)
  all.prom = reduce(all.prom, ignore.strand = TRUE)
  
  both = intersect(H3K4, prom)
  H3K4.only = setdiff(H3K4, both)
  prom.only = setdiff(prom, both)
  
  om = matrix(0, ncol = 2, nrow = 2)
  om[1,1] = sum(width(both))
  om[1,2] = sum(width(prom.only))
  om[2,1] = sum(width(H3K4.only))
  om[2,2] = 1.5*10^9 - sum(om)

  output = fisher.test(om)

  return(output)
}
```

Now, we can use the odds_ratio function to find our six odds ratios to fill a 2x3 matrix representing comparisons between the two expressed gene sets and the three H3K4me3 promoter genesets.

```{r}
ORMat <- matrix(0, ncol = 2, nrow = 3)
colnames(ORMat) <- c("fetal.prom", "adult.prom")
rownames(ORMat) <- c("fetal.H3K4", "adult.H3K4", "liver.H3K4")

PVMat <- matrix(0, ncol = 2, nrow = 3)
colnames(PVMat) <- c("fetal.prom", "adult.prom")
rownames(PVMat) <- c("fetal.H3K4", "adult.H3K4", "liver.H3K4")
```


```{r}
or11 = odds_ratio(fetal.H3K4.gr, fetal.prom, all.prom)
or12 = odds_ratio(fetal.H3K4.gr, adult.prom, all.prom)
or21 = odds_ratio(adult.H3K4.gr, fetal.prom, all.prom)
or22 = odds_ratio(adult.H3K4.gr, adult.prom, all.prom)
or31 = odds_ratio(liver.H3K4.gr, fetal.prom, all.prom)
or32 = odds_ratio(liver.H3K4.gr, adult.prom, all.prom)
```



```{r}
ORMat[1,1] = unname(or11$estimate)
ORMat[1,2] = unname(or12$estimate)
ORMat[2,1] = unname(or21$estimate)
ORMat[2,2] = unname(or22$estimate)
ORMat[3,1] = unname(or31$estimate)
ORMat[3,2] = unname(or32$estimate)
ORMat
```

```{r}
PVMat[1,1] = or11$p.value
PVMat[1,2] = or11$p.value
PVMat[2,1] = or11$p.value
PVMat[2,2] = or11$p.value
PVMat[3,1] = or11$p.value
PVMat[3,2] = or11$p.value
PVMat
```

