---
title: "Statistical Analysis"
author: "Harlan Gillespie"
date: "27/09/2021"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Dependencies

```{r Dependencies, message=FALSE}
library(SummarizedExperiment)
library(genefilter)
library(limma)
library(DESeq2)
library(ReportingTools)
library(dbplyr)
library(ggplot2)
```


## Load the data

Data is loaded  and the edata is transformed and filtered as it was in "Exploratory Data Analysis.Rmd".

```{r, echo=FALSE}
colData <- DataFrame(read.delim("~/Coursera Capstone/PData.txt", stringsAsFactors=TRUE))
counts <- as.matrix(read.delim("~/Coursera Capstone/featureCount-data-ENTREZ.txt"))
rownames(counts) = counts[,1]
counts = counts[,-1]
data.se = SummarizedExperiment(assays = list(counts = counts),  colData = colData)
edata = assays(data.se)[[1]]
pdata = colData(data.se)
filt.edata = edata[rowMeans(edata)>10,]
trans.edata = log2(filt.edata)+1
```



## Linear Regression

NOTE: Results from this section were not used. 

The null hypothesis is that there is no change in gene expression between fetal and adult (control) groups.

Model matrix is made to include both the variable of interest, Group, and the covariate RIN. RIN was shown to have some impact on gene expression in the PCA analysis so adjusting for this covariate is appropriate

```{r}
mod1 = model.matrix( ~ pdata$RIN + pdata$Group)
fit1 = lmFit(trans.edata, mod1)
fit1 = eBayes(fit1)
tt = topTable(fit1, number = Inf, coef = 3)
genes_logFC_LM = subset(tt, select = c(logFC, P.Value, adj.P.Val))

```

## Negative binomial regression using DESeq2

```{r}
de = DESeqDataSetFromMatrix(as.matrix(filt.edata), pdata, ~Group)
glm = DESeq(de)
result.nb = results(glm)
hist(result.nb$stat, breaks = 50)
result.df = as.data.frame(result.nb@listData)
rownames(result.df) = result.nb@rownames
genes_logFC = subset(result.df, select = c(log2FoldChange, pvalue, padj))
```


## Volcano Plot

Create labels for whether genes are up-regulated, down-regulated or neither. (+-)2 was chosen as a cut-off for logFC. P-values already take into account logFC so filtering by a high logFC is somewhat redundant. Also, a gene with an extremely high logFC does not necessarily suggest a greater effect on a biological function. A small change in expression of a gene on which many processes depend upon could have a greater effect on phenotype than a large change in expression of a gene involved in an island, isolated from significant networks. For this reason, filtering out too many genes with a low logFC value may exclude key findings.

```{r}
genes_logFC$diffexp = "NO"
genes_logFC$diffexp[genes_logFC$log2FoldChange > 2 & genes_logFC$padj < 0.05] = "UP"
genes_logFC$diffexp[genes_logFC$log2FoldChange < -2 & genes_logFC$padj < 0.05] = "DOWN"

```


Set colour scheme for differential expression.

```{r}
mycolors = c("blue", "red", "black")
names(mycolors) = c("DOWN", "UP", "NO")
```

Create new column which labels deferentially expressed genes

```{r}
genes_logFC$diffexplab = NA
genes_logFC$diffexplab[genes_logFC$diffexp != "NO"] = rownames(genes_logFC)[genes_logFC$diffexp != "NO"]

```


Create the volcano plot with colour coding and labels

```{r}

ggplot(data=genes_logFC, aes(x=log2FoldChange, y=-log10(padj), col=diffexp)) + 
    geom_point() +
    theme_minimal()
```


## Create TSV file 

Lastly, we can export the logFC, P.value and adjusted (FDR) P.value of each gene in the form of a TSV file.

```{r}
glFC_tsv = genes_logFC[,c(1,2,3)]
glFC_tsv = tibble::rownames_to_column(glFC_tsv, "Entrez.ID")
write.table(glFC_tsv, file = "genes_logFC.tsv", sep = "\t", col.names = colnames(glFC_tsv), row.names = FALSE)
```

## Session Info

```{r}
sessionInfo()
```








