---
title: "Statistical Analysis"
author: "Harlan Gillespie"
date: "27/09/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Dependencies

```{r Dependencies, message=FALSE}
library(SummarizedExperiment)
library(genefilter)
library(limma)
library(dbplyr)
library(ggplot2)
```


## Plot colour palette

```{r}
tropical=  c('darkorange', 'dodgerblue', 'hotpink', 'limegreen', 'yellow')
palette(tropical)
par(pch=19)
```



## Linear Regression

The null hypothesis is that there is no relationship between group and gene expression.

Model matrix is made to include both the variable of interest, Group, and the covariate RIN.

```{r}
mod1 = model.matrix( ~ pdata$RIN + pdata$Group)
fit1 = lmFit(filt_edata, mod1)
fit1 = eBayes(fit1)
tt = topTable(fit1, number = Inf, coef = 3)
genes_logFC = subset(tt, select = c(logFC, P.Value, adj.P.Val))

```



## Volcano Plot

Create labels for whether genes are upregulated, downregulated or neither.

```{r}
genes_logFC$diffexp = "NO"
genes_logFC$diffexp[genes_logFC$logFC > 1.5 & genes_logFC$adj.P.Val < 0.05] = "UP"
genes_logFC$diffexp[genes_logFC$logFC < -1.5 & genes_logFC$adj.P.Val < 0.05] = "DOWN"

```


Set colour scheme for differential expression.

```{r}
mycolors = c("blue", "red", "black")
names(mycolors) = c("DOWN", "UP", "NO")
```

Create new column which labels differentially expressed genes

```{r}
genes_logFC$diffexplab = NA
genes_logFC$diffexplab[genes_logFC$diffexp != "NO"] = rownames(genes_logFC)[genes_logFC$diffexp != "NO"]

```


Create the volcano plot with colour coding and labels

```{r}

ggplot(data=genes_logFC, aes(x=logFC, y=-log10(adj.P.Val), col=diffexp)) + 
    geom_point()
    theme_minimal()
```


## Create TSV file 

```{r}
glFC_tsv = genes_logFC[,c(1,2,3)]
glFC_tsv = tibble::rownames_to_column(glFC_tsv, "Entrez.ID")
write.table(glFC_tsv, file = "genes_logFC.tsv", sep = "\t", col.names = colnames(glFC_tsv), row.names = FALSE)
```












